storage/types.go

```diff
@@ -337,7 +337,27 @@ func (n *NodeID) Split(prefixBytes, suffixBits int) ([]byte, Suffix) {
 
 // Equivalent return true iff the other represents the same path prefix as this NodeID.
 func (n *NodeID) Equivalent(other NodeID) bool {
-	return n.String() == other.String()
+	// If they're different lengths they cannot represent the same path prefix.
+	if n.PrefixLenBits != other.PrefixLenBits {
+		return false
+	}
+
+	// The first depthBytes must be identical.
```

Comment:
You could potentially use `bytes.Equal` with slices into `Path` here

Reply:
Done.